addons/mail/models/mail_thread.py

```diff
@@ -1150,6 +1222,10 @@ def message_new(self, msg_dict, custom_values=None):
         name_field = self._rec_name or 'name'
         if name_field in fields and not data.get('name'):
             data[name_field] = msg_dict.get('subject', '')
+
+        if self._primary_email and self._primary_email in self._fields and msg_dict.get('email_from'):
```

Comment:
Note: we could define ``_assert_primary_email`` a bit like in mail.thread.blacklist, exception would be that if _primary_email is not defined or Falsy, we just return False instead of raising. Would maybe then name it ``_check_primary_email`` if it does not always assert d'ailleurs . That way as I bet we will use more this primary email in the future, we would have a small helper instead of always having to copy-paste this line :) .